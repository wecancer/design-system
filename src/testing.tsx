import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
