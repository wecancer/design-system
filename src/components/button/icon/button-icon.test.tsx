import ButtonIcon from '.'
import { render, fireEvent } from '../../../testing'

describe('<ButtonIcon />', () => {
  it('should render snapshot correctly', () => {
    const { container } = render(
      <>
        <ButtonIcon icon="search" />
        <ButtonIcon color="error" icon="minus" />
        <ButtonIcon title="title test" icon="view" />
        <ButtonIcon title="title test" icon="view" isDisabled />
        <ButtonIcon icon="view" isLoading />
      </>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should add the className in button component', () => {
    const { container } = render(
      <ButtonIcon className="customClass" icon="check" />,
    )
    expect(container.querySelector('.customClass')).toBeInstanceOf(
      HTMLButtonElement,
    )
  })

  it('should receive the isDisabled attribute', () => {
    const handleClick = jest.fn()
    const { container } = render(
      <ButtonIcon icon="check" onClick={handleClick} isDisabled />,
    )
    const btnElement = container.querySelector('button')

    if (!btnElement) throw new Error("Can't find the icon button")

    fireEvent.click(btnElement)
    expect(handleClick).toHaveBeenCalledTimes(0)
  })

  it('should receive the isLoading attribute', () => {
    const { container } = render(<ButtonIcon icon="check" isLoading />)
    expect(
      container.querySelector('[data-icon-name="spinnerSolid"]'),
    ).toBeDefined()
  })

  it('should receive the onChange attribute', () => {
    const handleClick = jest.fn()
    const { container } = render(
      <ButtonIcon icon="check" onClick={handleClick} />,
    )
    const btnElement = container.querySelector('button')

    if (!btnElement) throw new Error("Can't find the icon button")

    fireEvent.click(btnElement)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })
})
