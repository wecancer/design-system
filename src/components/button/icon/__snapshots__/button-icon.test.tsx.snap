// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonIcon /> should render snapshot correctly 1`] = `
<div>
  <button
    class="sc-dkPtRN ftGamc"
  >
    <div
      class="sc-bdvvtL bYGNWJ"
      data-icon-name="search"
    >
      <svg
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 20C15.9706 20 20 15.9706 20 11C20 6.02944 15.9706 2 11 2C6.02944 2 2 6.02944 2 11C2 15.9706 6.02944 20 11 20Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <path
          d="M22 22L18 18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </button>
  <button
    class="sc-dkPtRN kUqCeI"
  >
    <div
      class="sc-bdvvtL bYGNWJ"
      data-icon-name="minus"
    >
      <svg
        fill="none"
        viewBox="0 0 21 2"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.79199 1L20.0003 1"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </button>
  <button
    class="sc-dkPtRN ftGamc"
    title="title test"
  >
    <div
      class="sc-bdvvtL bYGNWJ"
      data-icon-name="view"
    >
      <svg
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.33497 13.2561C0.888345 12.4782 0.888342 11.522 1.33497 10.7441C3.68496 6.65097 7.44378 4 11.6798 4C15.9158 4 19.6746 6.65094 22.0246 10.744C22.4712 11.5219 22.4712 12.4781 22.0246 13.256C19.6746 17.3491 15.9158 20 11.6798 20C7.44377 20 3.68497 17.3491 1.33497 13.2561Z"
          stroke="currentColor"
          stroke-width="2"
        />
        <circle
          cx="11.6797"
          cy="12"
          r="3"
          stroke="currentColor"
          stroke-width="2"
        />
      </svg>
    </div>
  </button>
  <button
    class="sc-dkPtRN ftGamc"
    disabled=""
    title="title test"
  >
    <div
      class="sc-bdvvtL bYGNWJ"
      data-icon-name="view"
    >
      <svg
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.33497 13.2561C0.888345 12.4782 0.888342 11.522 1.33497 10.7441C3.68496 6.65097 7.44378 4 11.6798 4C15.9158 4 19.6746 6.65094 22.0246 10.744C22.4712 11.5219 22.4712 12.4781 22.0246 13.256C19.6746 17.3491 15.9158 20 11.6798 20C7.44377 20 3.68497 17.3491 1.33497 13.2561Z"
          stroke="currentColor"
          stroke-width="2"
        />
        <circle
          cx="11.6797"
          cy="12"
          r="3"
          stroke="currentColor"
          stroke-width="2"
        />
      </svg>
    </div>
  </button>
  <button
    class="sc-dkPtRN ftGamc"
  >
    <div
      class="sc-gsDKAQ cbKTir"
    >
      <div
        class="sc-bdvvtL bYGNWJ"
        data-icon-name="spinnerSolid"
      >
        <svg
          fill="none"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M10 1.67A8.33 8.33 0 1018.33 10"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
          />
        </svg>
      </div>
    </div>
  </button>
</div>
`;
